app/lucy/src/app/components/admin-tools/admin-tools.component.ts

```diff
@@ -13,112 +12,41 @@ import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
 })
 
 export class AdminToolsComponent implements OnInit {
-  public requests: accessRequest[] = []
-  public allUsers: usersList[] = []
-  public activeRoles: userRole[] = []
+  public requests: AccessRequest[] = []
+  public allUsers: User[] = []
+  public activeRoles: Role[] = []
 
-  public focusedAccessRequest: accessRequest = {
-    id: 0,
-    username: "",
-    name: "",
-    currentRole: "",
-    requestedRole: "",
-    reasons: "",
+  public focusedAccessRequest: AccessRequest;
 
-    responseRole: "",
-    responseMessage: "",
-  }
-
-  constructor(private roles: RolesService, private formsModule: FormsModule) { }
-
-  ngOnInit() {
-    this.roles.getAllActiveRoles().then((value) => {
-      this.activeRoles = value;
-    });
-    this.getDummyRequests().then((value) => {
-      this.requests = value;
-    });
-    this.getDummyUserslist().then((value) => {
-      this.allUsers = value;
-    });
-  }
+  constructor(private roles: RolesService, private admin: AdminService) { }
 
-  public setFocusedAccessRequest(request: accessRequest) {
-    this.focusedAccessRequest = request;
-    console.log("request for modal set");
-  }
+  ngOnInit() {}
 
-  public removeUser(user: usersList) {
-    console.log(user.role);
-    console.log(user.isActive);
-    console.log("******")
-    console.log("TODO: Make api call to Remove User user:")
-    console.log(user)
-    console.log("******")
+  ngAfterViewInit() {
+    this.getAllRoles();
+    this.getAllUsers();
+    this.getAllRequests();
   }
 
-  public setUserRole(user: usersList, role: userRole) {
-    user.role = role.name
-    console.log("******")
-    console.log("TODO: Make api call to set role on user:")
-    console.log(user)
-    console.log("New Role:")
-    console.log(role)
-    console.log("******")
+  private getAllRequests() {
+    this.admin.getRequests().then((value) => {
+      this.requests = value
+    });
   }
 
-  public userActiveStatusChanged(user: usersList, isActive: boolean) {
-    console.log("******")
-    console.log("TODO: Make api call to Activate/DeActivate user access:")
-    console.log(user)
-    console.log("Account status active:")
-    console.log(isActive)
-    console.log("******")
+  private getAllUsers() {
+    this.admin.getAllUsers().then((value) => {
+      this.allUsers = value
+    });
   }
```

Comment:
you can make it as async like
private async getAllUser: Promise<void> => {
    const users = await this.admin.getAllUsers();
    this.allUsers = users;
    return;
}